### PR TITLE
feat: [ADS 2.0] First time onboarding Build your own app flow

### PR DESCRIPTION
**Pull request by rohitagarwal88**
https://www.loom.com/share/eb07bde31d2243679a1df35efe5cf9df

Fixes #[22654](https://github.com/appsmithorg/appsmith/issues/22654)

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update

- Manual
- Jest
- Cypress

### Issues raised during DP testing
https://github.com/appsmithorg/appsmith/pull/22645#issuecomment-1519653605

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx [hunk 1]
```diff
@@ -78,65 +73,42 @@ const StatusWrapper = styled.p`
...
-  cursor: pointer;
-`;
-
 const Banner = styled.div`
   width: calc(100% - 113px);
-  border: 1px solid ${(props) => props.theme.colors.table.border};
-  padding: ${(props) => props.theme.spaces[7]}px;
-  margin-top: ${(props) => props.theme.spaces[7]}px;
+  border: 1px solid var(--ads-v2-color-border);
+  padding: var(--ads-v2-spaces-7);
```
Comment (albinAppsmith) [inline app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx:110]: `--ads-v2-spaces-7` is `24px` and `props.theme.spaces[7]` is `16px`. Is this intentional?

File: app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx [hunk 2]
```diff
@@ -297,25 +262,17 @@ export default function OnboardingChecklist() {
...
-            </CompeleteMarkerIcon>
+            <Icon
+              className="flex"
+              color={
+                datasources.length || actions.length
+                  ? "var(--ads-v2-color-fg-success)"
+                  : ""
+              }
+              data-testid="checklist-datasource-complete-icon"
+              name="oval-check"
```
Comment (albinAppsmith) [inline app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx:273]: There was two icons
`icon={
                  datasources.length || actions.length
                    ? "tick-circle"
                    : "small-tick"
                }`

Why this is changed to one(`oval-check`)?

File: app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx [hunk 3]
```diff
@@ -356,19 +314,13 @@ export default function OnboardingChecklist() {
...
-                data-testid="checklist-action-complete-icon"
-                icon={actions.length ? "tick-circle" : "small-tick"}
-                iconSize={17}
-              />
-            </CompeleteMarkerIcon>
+            <Icon
+              className="flex"
+              color={actions.length ? "var(--ads-v2-color-fg-success)" : ""}
+              data-testid="checklist-action-complete-icon"
+              name="oval-check"
```
Comment (albinAppsmith) [inline app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx:321]: Here also same, there was two icons before

File: app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx [hunk 4]
```diff
@@ -398,28 +350,25 @@ export default function OnboardingChecklist() {
...
-            </CompeleteMarkerIcon>
+            <Icon
+              className="flex"
+              color={
+                Object.keys(widgets).length > 1
+                  ? "var(--ads-v2-color-fg-success)"
+                  : ""
+              }
+              data-testid="checklist-widget-complete-icon"
+              name="oval-check"
```
Comment (albinAppsmith) [inline app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx:369]: Here as well

File: app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx [hunk 5]
```diff
@@ -443,26 +392,23 @@ export default function OnboardingChecklist() {
...
-                iconSize={17}
-              />
-            </CompeleteMarkerIcon>
+            <Icon
+              className="flex"
+              color={
+                isConnectionPresent ? "var(--ads-v2-color-fg-success)" : ""
+              }
+              data-testid="checklist-connection-complete-icon"
+              name="oval-check"
```
Comment (albinAppsmith) [inline app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx:409]: +

File: app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx [hunk 6]
```diff
@@ -494,17 +441,13 @@ export default function OnboardingChecklist() {
...
-                icon={isDeployed ? "tick-circle" : "small-tick"}
-                iconSize={17}
-              />
-            </CompeleteMarkerIcon>
+            <Icon
+              className="flex"
+              color={isDeployed ? "var(--ads-v2-color-fg-success)" : ""}
+              data-testid="checklist-deploy-complete-icon"
+              name="oval-check"
+              size="lg"
```
Comment (albinAppsmith) [inline app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx:449]: +here

File: app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx (hunk 2 above)
Comment (rohitagarwal88) [reply app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx:273]: The old implementation had an `icon` inside the `div`, the div had a `border-color` which changes based on task was completed or not.

Refactored implementation removes `div` & use single `icon` & changes icon color based on whether task was completed or not.

File: app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx (hunk 3 above)
Comment (rohitagarwal88) [reply app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx:321]: The old implementation had an icon inside the div, the div had a border-color which changes based on task was completed or not.

Refactored implementation removes div & use single icon & changes icon color based on whether task was completed or not.

File: app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx (hunk 4 above)
Comment (rohitagarwal88) [reply app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx:369]: The old implementation had an icon inside the div, the div had a border-color which changes based on task was completed or not.

Refactored implementation removes div & use single icon & changes icon color based on whether task was completed or not.

File: app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx (hunk 5 above)
Comment (rohitagarwal88) [reply app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx:409]: The old implementation had an icon inside the div, the div had a border-color which changes based on task was completed or not.

Refactored implementation removes div & use single icon & changes icon color based on whether task was completed or not.

File: app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx (hunk 6 above)
Comment (rohitagarwal88) [reply app/client/src/pages/Editor/FirstTimeUserOnboarding/Checklist.tsx:449]: @albinAppsmith 
The old implementation had an icon inside the div, the div had a border-color which changes based on task was completed or not.

Refactored implementation removes div & use single icon & changes icon color based on whether task was completed or not.

**Comment by shadabbuchh** [comment]
Issues noticed in this PR:
https://www.notion.so/appsmith/f353d8c6bd664f79ad858a42010cdfc8?v=f04cde23f6424aeb9d5a6e389cd172bd&p=c91aa60666b2488db5b0b0abc0321ab9&pm=s

Please note that the Design audit for this has not yet been done.

**Review by akash-codemonk — APPROVED** [review]

**Review by albinAppsmith — APPROVED** [review]

**Pull request merged**